lib/ClangImporter/ImportDecl.cpp

```diff
@@ -9479,15 +9488,18 @@ void ClangImporter::Implementation::loadAllMembersOfRecordDecl(
   for (auto member: members) {
     // This means we found a member in a C++ record's base class.
     if (swiftDecl->getClangDecl() != clangRecord) {
+      auto namedMember = cast<ValueDecl>(member);
+      if (namedMember->getFormalAccess() < AccessLevel::Public)
```

Comment:
Could we run into similar problems when of spuriously marking a class sendable if we do not import add a field from the base class?

Reply:
Yeah, this change isn't enough to fix the inheritance case, since Sema's `StorageVisitor` only considers stored properties, while for inherited C++ types we synthesize a computed property. Let me see if we can change Sema logic to accommodate this.